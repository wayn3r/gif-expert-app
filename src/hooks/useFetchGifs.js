import { useEffect, useState } from 'react'
import { getGifs } from '../helpers'

export const useFetchGifs = category => {
    const [state, setState] = useState({ gifs: [], loading: true })
    useEffect(() => {
        setState(({ gifs }) => ({ loading: true, gifs }))
        getGifs(category).then(gifs => setState({ loading: false, gifs }))
    }, [category])
    return state
}
