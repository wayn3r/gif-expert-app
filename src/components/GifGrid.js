import { useFetchGifs } from '../hooks'
import GifGridItem from './GifGridItem'
import PropTypes from 'prop-types'
const GifGrid = ({ category }) => {
    const { gifs, loading } = useFetchGifs(category)

    return (
        <div className="card-grid">
            <h3 className="card-title animate__animated animate__fadeIn">
                {category}
            </h3>
            {loading ? (
                <p className="animate__animated animate__flash">Cargando...</p>
            ) : (
                gifs.map(({ id, ...img }) => <GifGridItem key={id} {...img} />)
            )}
        </div>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
export default GifGrid
