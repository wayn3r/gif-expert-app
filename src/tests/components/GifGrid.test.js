import { shallow } from 'enzyme'
import GifGrid from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas al componente <GifGrid />', () => {
    const category = 'Naruto'
    useFetchGifs.mockReturnValue({
        gifs: [],
        loading: true,
    })
    const wrapper = shallow(<GifGrid category={category} />)
    test('debe cargar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })
    test('debe mostrar items cuando se carga las imagenes', () => {
        const gifs = [
            {
                id: 'id de la imagen',
                title: 'title de la image',
                url: 'downsized_medium url de la imagen',
            },
            {
                id: 'id de la imagen 2',
                title: 'title de la image',
                url: 'downsized_medium url de la imagen',
            },
        ]
        useFetchGifs.mockReturnValue({
            gifs,
            loading: false,
        })
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
})
