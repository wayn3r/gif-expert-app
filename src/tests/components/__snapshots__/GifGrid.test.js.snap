// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifGrid /> debe cargar el componente correctamente 1`] = `
<div
  className="card-grid"
>
  <h3
    className="card-title animate__animated animate__fadeIn"
  >
    Naruto
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
</div>
`;
